rat/grammar.ebnf

```diff
@@ -0,0 +1,56 @@
+@@grammar::Rat
+@@eol_comments :: /#.*?$/
+@@keyword :: let in end fun
+@@parseinfo :: True
+
+start::Program = { statements+:statement }+ $;
+
+statement::Statement = left:factor ~ op:('~' | '=') ~ right:expression ';';
```

Comment:
Hrmm, I maybe just try promoting the left hand side to an expression? I think you hit a problem when the left-most-this is kinda recursively defined.

Like this would be a problem:
```
statement = left:statement '~' right:statement
```

I'm not really sure what the rules are but you can see that in the mini calc example: https://tatsu.readthedocs.io/en/stable/mini-tutorial.html

I think we can't assign to expressions tho, so this'd need to break out into one rule for `~` and one rule for `=`.

Reply:
Yeah we do need to separate them.

Reply:
I'll open an issue for this since I dont want it to get lost.